src/fty_security_wallet.cc

```diff
@@ -107,45 +107,47 @@ int main (int argc, char *argv [])
         }
 
         log_info(SECURITY_WALLET_AGENT " starting");
-        
+
         //create params for SECW
         Arguments paramsSecw;
-        
+
         paramsSecw["STORAGE_CONFIGURATION_PATH"] = storage_access_path;
         paramsSecw["STORAGE_DATABASE_PATH"] = storage_database_path;
         paramsSecw["AGENT_NAME"] = secw_actor_name;
         paramsSecw["ENDPOINT"] = endpoint;
         paramsSecw["AGENT_NAME_SRR"] = secw_actor_name;
         paramsSecw["ENDPOINT_SRR"] = endpoint;
-        
+
         //start secw agent
-        
+
         //create a stream publisher for notification
         mlm::MlmStreamClient notificationStream(SECURITY_WALLET_AGENT, SECW_NOTIFICATIONS, 1000, paramsSecw.at("ENDPOINT"));
-    
+
         //create the server
         secw::SecurityWalletServer serverSecw(  paramsSecw.at("STORAGE_CONFIGURATION_PATH"),
                                         paramsSecw.at("STORAGE_DATABASE_PATH"),
                                         notificationStream,
                                         paramsSecw.at("ENDPOINT_SRR"),
                                         paramsSecw.at("AGENT_NAME_SRR"));
-        
+
         fty::SocketBasicServer agentSecw( serverSecw, socketPath);
 
+#if defined (HAVE_LIBSYSTEMD)
+        //notify systemd that the socket is ready, so that depending units can start
+        sd_notify(0, "READY=1");
+#endif
```

Comment:
Shouldn't the notification be sent after the agent has fully started up (i.e. just before entering its mainloop)?

Reply:
Probably, though this question I would delegate to @aquette and PR #49 ;)

Reply:
Indeed, posted the improvement later in PR development.